scripts/bring_assisted_service.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+export SERVICE_REPO=${SERVICE_REPO:-https://github.com/openshift/assisted-service}
+export SERVICE_BRANCH=${SERVICE_BRANCH:-master}
+export SERVICE_BASE_REF=${SERVICE_BASE_REF:-master}
+export OPENSHIFT_CI=${OPENSHIFT_CI:-false}
+export REPO_NAME=${REPO_NAME:-assisted-service}
+export JOB_TYPE=${JOB_TYPE:-}
+export PULL_BASE_REF=${PULL_BASE_REF:-master}
+
+if ! [[ -d "assisted-service" ]]; then
+  echo "Can't find assisted-service source locally, cloning ${SERVICE_REPO}"
+  git clone "${SERVICE_REPO}"
+fi
+
+service_active_branch=$(cd assisted-service/ && git rev-parse --abbrev-ref HEAD)
+pr_branch_name=assisted-service-pr-${PULL_NUMBER}
+
+if [[ "${OPENSHIFT_CI}" == "true" && "${REPO_NAME}" == "assisted-service" && "${JOB_TYPE}" == "presubmit" ]]; then
+  if [[ "${service_active_branch}" == "${pr_branch_name}" ]]; then
+    echo "Nothing to update. ${REPO_NAME} is already updated, branch: ${pr_branch_name}"
```

Comment:
Why do we know there is no need to rebase here?

Reply:
It fails when we call it twice (on different places on the code). If the branch is already exists it means that we already fetched, checked-out and rebsaed the code

Reply:
I guess it fails on the `checkout` command, you can mitigate it by adding `-B` to the command (it will reset the branch in case it already exists).

If you decide to keep it this way please add a comment on why we do this check and that what it means

Reply:
It's failing when trying to run the command 
```bash
git fetch -v origin pull/3678/head:assisted-service-pr-3678
--
fatal: Refusing to fetch into current branch refs/heads/assisted-service-pr-3678 of non-bare repository
```

Reply:
After the rebase it fails even with this flag.
With the flag there is another failure:

```
From https://github.com/openshift/assisted-service
 ! [rejected]          refs/pull/3678/head -> assisted-service-pr-3678  (non-fast-forward)
```

Reply:
And from the documentation the recommendation is not to use the `--update-head-ok` flag

```
--update-head-ok
By default git fetch refuses to update the head which corresponds to the current branch. This flag disables the check. This is purely for the internal use for git pull to communicate with git fetch, and unless you are implementing your own Porcelain you are not supposed to use it.
```

Reply:
I documented it